dhall-docs/src/Dhall/Docs.hs

```diff
@@ -51,20 +68,116 @@ parseOptions :: Parser Options
 parseOptions =
         Options
     <$> Options.Applicative.strOption
-        ( Options.Applicative.long "path"
-       <> Options.Applicative.metavar "PATH"
-       <> Options.Applicative.help "Root folder of your dhall package" )
+        ( Options.Applicative.long "input"
+       <> Options.Applicative.metavar "INPUT"
+       <> Options.Applicative.help "Directory of your dhall package. It should \
+                                    \contain only text files" )
     <*> parseStrategy
+    <*> Options.Applicative.strOption
+        ( Options.Applicative.long "output"
+       <> Options.Applicative.metavar "OUTPUT"
+       <> Options.Applicative.help "Directory where your docs will be generated"
+       <> Options.Applicative.value "docs" )
 
 -- | `ParserInfo` for the `Options` type
 parserInfoOptions :: ParserInfo Options
 parserInfoOptions =
+    let progDesc = "Generate HTML documentation from a dhall package or file" in
     Options.Applicative.info
         (Options.Applicative.helper <*> parseOptions)
         (   Options.Applicative.fullDesc
-        <>  Options.Applicative.progDesc "Generate HTML documentation from a dhall package or file"
+        <>  Options.Applicative.progDesc progDesc
         )
 
+{-| Fetches a list of all dhall files in a directory. This is not the same
+    as finding all files that ends in @.dhall@, but finds all files that
+    successfully parses as a valid dhall file.
+
+    The reason it doesn't guide the search by its extension is because of the
+    dhall @Prelude@ (<https://github.com/dhall-lang/dhall-lang/tree/master/Prelude>)
+    That package doesn't ends any of their files in @.dhall@.
+
+    TODO: Avoid processing images
+-}
+getAllDhallFiles
+    :: FilePath -- ^ Base directory to do the search
+    -> IO [(FilePath, Header)]
+getAllDhallFiles baseDir = do
+
+    let shell = do
+            path_ <- Turtle.lstree baseDir
+            isNotDir <- not <$> Turtle.testdir path_
+            Control.Monad.guard isNotDir
+
+            let pathStr = Text.unpack $ Turtle.format fp path_
+            contents <- Turtle.liftIO $ Text.IO.readFile pathStr
+
+            case exprAndHeaderFromText pathStr contents of
+                Right (header, _) -> return (path_, header)
+                _ -> Turtle.empty
+
+    Turtle.fold shell Foldl.list
+
+{-| Calculate the relative path needed to access files on the first argument
+    relative from the second argument.
+
+    The second argument needs to be a child of the first, otherwise it will
```

Comment:
I think a more reliable way to do this is to make the two paths absolute, delete the longest shared prefix between the two paths, and use the remainder to compute the relative path between the two